docs/developer/External-Api.md

```diff
@@ -0,0 +1,810 @@
+* All string fields in a JSON response **expect an item's body** are **not sanitized**. This means that if you do not escape it properly before rendering you will be vulnerable to [XSS](https://www.owasp.org/index.php/Cross-site_Scripting_%28XSS%29) attacks
+* Basic Auth headers can easily be decrypted by anyone since base64 is an encoding, not an encryption. Therefore only send them if you are accessing an HTTPS website or display an easy to understand warning if the user chooses HTTP
+* When creating a feed you can choose to add basic auth authentication credentials. These must be stored in clear text so anyone with access to your database (however they might have achieved it, think of Sql injection) can read them and use them to access the website. You should warn the user about this.
+* If you are building a client in JavaScript or are using a link with **target="blank"**, remember to set the **window.opener** property to **null** and/or add a **rel="noreferrer"** to your link to prevent your app from being [target by an XSS attack](https://medium.com/@jitbit/target-blank-the-most-underestimated-vulnerability-ever-96e328301f4c#.wf2ddytbh)
+
+## Syncing
+All routes are given relative to the base API url, e.g.: **/sync** becomes  **https://yourowncloud.com/index.php/apps/news/api/v2/sync**
+
+There are two usecases for syncing:
+* **Initial sync**: the user does not have any data at all
+* **Syncing local and remote changes**: the user has synced at least once and wants to submit and receive changes
+
+### Initial Sync
+The intial sync happens when a user adds an ownCloud account in your app. In that case you want to download all folders, feeds and unread/starred items. To do this, make the following request:
+
+* **Method**: GET
+* **Route**: /sync
+* **Authentication**: [required](#authentication)
+* **HTTP headers**:
+  * **Accept: "application/json"**
+
+This will return the following status codes:
+* **200**: Success
+
+and the following HTTP headers:
+* **Content-Type**: application/json; charset=utf-8
+* **Etag**: A string containing a cache header, maximum size 64 ASCII characters, e.g. 6d82cbb050ddc7fa9cbb659014546e59
+
+and the following request body:
+```js
+{
+    "folders": [ /* array of folder objects */ ],
+    "feeds": [ /* array of feed objects */ ],
+    "items": [ /* array of item objects */ ]
+}
+```
+
+**Note**: Each object is explained in more detail in a separate section:
+* [Folders](#folders)
+* [Feeds](#feeds)
+* [Items](#items)
+
+
+### Sync Local And Remote Changes
+After the initial sync the app has all folders, feeds and items. Now you want to push changes and retrieve updates from the server. To do this, make the following request:
+
```

Comment:
It's unclear how locally or remotely added or deleted feeds and folders are represented in this scenario. Can you elaborate on that?